bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -1527,3 +1590,100 @@ impl MediaFileHandle {
         )
     }
 }
+
+#[cfg(test)]
```

Comment:
More of a question for the Rust team, but I thought we didn't want to test anything in the FFI and if any logic needed testing it should instead live in the main SDK or UI crate, to keep the FFI down to lightweight wrappers.

Reply:
Oh, I've only added these tests because of @Hywan's comment [above](https://github.com/matrix-org/matrix-rust-sdk/pull/3558#pullrequestreview-2122770915).

Reply:
Ideally if something needs tests, it should live in a `crates/` crate, not in FFI. FFI bindings should be as thin as possible with no logic except mapping data from the foreign language to Rust and vice versa.

Reply:
Have pushed the callback URL parsing and tests down into the SDK with b8f69b0ed98942a4b7ca09a38f97d0d4a157e584.